### PR TITLE
Fly to: stay in ortho when in 2d mode

### PR DESCRIPTION
**Pull request by nd-novorender**
https://trello.com/c/Ns66jI24/813-fly-to-2d-switches-to-3d

### CODE REVIEW

File: src/features/flyToSelected/flyToSelected.tsx [hunk 1]
```diff
@@ -42,13 +47,49 @@ export function FlyToSelected({ position, ...speedDialProps }: Props) {
...
+                let radius = sphere.radius;
+                if (isCrossSection && view.renderState.clipping.planes.length > 0) {
+                    // In cross section try to reduce bounding sphere radius based on
+                    // the clipping plane position (use sphere/plane intersection radius)
+                    // because visible object part might be much smaller than the full bounding sphere
+                    const plane = vec4.copy(vec4.create(), view.renderState.clipping.planes[0].normalOffset);
+                    vec4.negate(plane, plane);
+                    plane[3] = -plane[3];
+                    const dist = pointToPlaneDistance(sphere.center, plane);
+                    if (dist > 1e-6 && dist < sphere.radius) {
```
Comment (barabadzhi) [inline src/features/flyToSelected/flyToSelected.tsx:70]: Is the `1e-6` empirical or ..?

File: src/features/flyToSelected/flyToSelected.tsx [hunk 2]
```diff
@@ -1,23 +1,31 @@
...
+import { isRectInsideCircle, pointToPlaneDistance, pointToRectDistance } from "utils/math";
 import { objIdsToTotalBoundingSphere } from "utils/objectData";
+import { sleep } from "utils/time";
 
 enum Status {
     Initial,
     Loading,
 }
 
+const ORTHO_PADDING = 0.2;
```
Comment (barabadzhi) [inline src/features/flyToSelected/flyToSelected.tsx:27]: Origin of the constant / reasoning behind this value?
Please, add a comment.

File: src/features/flyToSelected/flyToSelected.tsx [hunk 3]
```diff
@@ -29,26 +37,159 @@ export function FlyToSelected({ position, ...speedDialProps }: Props) {
...
+    const viewMode = useAppSelector(selectViewMode);
+    const isCrossSection = [ViewMode.CrossSection, ViewMode.FollowPath, ViewMode.Deviations].includes(viewMode);
 
     const [status, setStatus] = useState(Status.Initial);
 
     const previousBoundingSphere = useRef<BoundingSphere>();
     const [abortController, abort] = useAbortController();
 
+    const [tooltipMessage, setTooltipMessage] = useState("");
+    const tooltipTimeout = useRef<ReturnType<typeof setTimeout>>();
```
Comment (barabadzhi) [inline src/features/flyToSelected/flyToSelected.tsx:50]: Isn't the return of `setTimeout` just a `number` in the browser env?

File: src/features/flyToSelected/flyToSelected.tsx [hunk 4]
```diff
@@ -29,26 +37,159 @@ export function FlyToSelected({ position, ...speedDialProps }: Props) {
...
+        if (orthoLoadingHandle.current) {
+            setOrthoLoading(false);
+            orthoLoadingHandle.current = 0;
+        }
+
+        const lookAtSphereInOrtho = (sphere: BoundingSphere) => {
+            const cameraDir = getCameraDir(view.renderState.camera.rotation);
+            const pinholeFov = view.controllers.flight.fov;
+            let dist = Math.max(sphere.radius / Math.tan(glMatrix.toRadian(pinholeFov) / 2), sphere.radius);
+            dist = Math.min(dist, view.renderState.camera.far * 0.9);
```
Comment (barabadzhi) [inline src/features/flyToSelected/flyToSelected.tsx:84]: Are both of them `dist` really?
Or is the first one some kind of intermediate? 🤔

File: src/features/flyToSelected/flyToSelected.tsx [hunk 5]
```diff
@@ -29,26 +37,159 @@ export function FlyToSelected({ position, ...speedDialProps }: Props) {
...
+                            },
+                        })
+                    );
+                }
+            } else {
+                lookAtSphereInOrtho(sphere);
+            }
+        };
+
+        const go = (sphere: BoundingSphere) => {
```
Comment (barabadzhi) [inline src/features/flyToSelected/flyToSelected.tsx:182]: I'd advocate for name change to `goTo` in line with other `goToX` fns.

File: src/features/flyToSelected/flyToSelected.tsx (hunk 3 above)
Comment (nd-novorender) [reply src/features/flyToSelected/flyToSelected.tsx:50]: My VS code seems to pick NodeJS return type for this particular setTimeout usage, so this type makes it so I don't have to care about return type

File: src/features/flyToSelected/flyToSelected.tsx (hunk 4 above)
Comment (nd-novorender) [reply src/features/flyToSelected/flyToSelected.tsx:84]: Second one is dist constrained by a maximum value, I'll call it `offset`

**Review by barabadzhi — APPROVED** [review]
LGTM

**Pull request merged**